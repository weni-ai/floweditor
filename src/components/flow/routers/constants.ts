import i18n from 'config/i18n';

export const WAIT_LABEL = i18n.t(
  'forms.message_label',
  'If the message response...',
);
export const EXPRESSION_LABEL = i18n.t(
  'forms.expression_label',
  'If the expression...',
);
export const GROUP_LABEL = i18n.t(
  'forms.split_by_groups',
  "Select the groups you'd like to split by below",
);

export const OPERAND_LOCALIZATION_DESC = i18n.t(
  'forms.localize_rules',
  'Sometimes languages need special rules to route things properly. If a translation is not provided, the original rule will be used.',
);

// We intentionally don't localize these names
export enum DefaultExitNames {
  All_Responses = 'All Responses',
  No_Response = 'No Response',
  Any_Value = 'Any Value',
  Other = 'Other',
}
