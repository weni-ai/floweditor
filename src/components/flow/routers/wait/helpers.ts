import {
  createRenderNode,
  resolveRoutes,
} from 'components/flow/routers/helpers';
import { WaitRouterFormState } from 'components/flow/routers/wait/WaitRouterForm';
import {
  DEFAULT_OPERAND,
  MEDIA_OPERAND,
} from 'components/nodeeditor/constants';
import { Type, Types } from 'config/interfaces';
import {
  HintTypes,
  Router,
  RouterTypes,
  SwitchRouter,
  Wait,
  WaitTypes,
} from 'flowTypes';
import { RenderNode } from 'store/flowContext';
import { NodeEditorSettings, StringEntry } from 'store/nodeEditor';

export const nodeToState = (
  settings: NodeEditorSettings,
): WaitRouterFormState => {
  let resultName: StringEntry = { value: 'Result' };

  // look at the actual ui type instead of the hint here, we want results for any kind of wait
  if (
    settings.originalNode &&
    settings.originalNode.ui.type === Types.wait_for_response
  ) {
    const router = settings.originalNode.node.router as SwitchRouter;
    if (router) {
      resultName = { value: router.result_name || '' };
    }
  }

  return {
    resultName,
    valid: true,
  };
};

export const stateToNode = (
  settings: NodeEditorSettings,
  state: WaitRouterFormState,
  typeConfig: Type,
): RenderNode => {
  const {
    exits,
    defaultCategory: defaultExit,
    caseConfig,
    categories,
  } = resolveRoutes([], false, settings.originalNode.node);

  const optionalRouter: Pick<Router, 'result_name'> = {};
  if (state.resultName.value) {
    optionalRouter.result_name = state.resultName.value;
  }

  let operand = DEFAULT_OPERAND;
  const wait = { type: WaitTypes.msg } as Wait;
  switch (typeConfig.type) {
    case Types.wait_for_audio:
      wait.hint = { type: HintTypes.audio };
      operand = MEDIA_OPERAND;
      break;
    case Types.wait_for_image:
      wait.hint = { type: HintTypes.image };
      operand = MEDIA_OPERAND;
      break;
    case Types.wait_for_location:
      wait.hint = { type: HintTypes.location };
      operand = MEDIA_OPERAND;
      break;
    case Types.wait_for_video:
      wait.hint = { type: HintTypes.video };
      operand = MEDIA_OPERAND;
      break;
  }

  const router: SwitchRouter = {
    type: RouterTypes.switch,
    default_category_uuid: defaultExit,
    cases: [],
    categories,
    wait,
    operand,
    ...optionalRouter,
  };

  const newRenderNode = createRenderNode(
    settings.originalNode.node.uuid,
    router,
    exits,
    Types.wait_for_response,
    [],
    { cases: caseConfig },
  );

  return newRenderNode;
};
