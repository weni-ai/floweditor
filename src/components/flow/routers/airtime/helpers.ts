import {
  AirtimeRouterFormState,
  AirtimeTransferEntry,
} from 'components/flow/routers/airtime/AirtimeRouterForm';
import { createWebhookBasedNode } from 'components/flow/routers/helpers';
import { Types } from 'config/interfaces';
import { TransferAirtime } from 'flowTypes';
import { RenderNode } from 'store/flowContext';
import { NodeEditorSettings } from 'store/nodeEditor';
import { createUUID } from 'utils';

export const nodeToState = (
  settings: NodeEditorSettings,
): AirtimeRouterFormState => {
  const originalAction = getOriginalAction(settings);
  let resultName = { value: 'Result' };
  let valid = false;

  const amounts: AirtimeTransferEntry[] = [];
  if (originalAction && originalAction.type === Types.transfer_airtime) {
    Object.keys(originalAction.amounts).forEach((key: string) => {
      amounts.push({
        value: { code: key, amount: '' + originalAction.amounts[key] },
      });
    });
    resultName = { value: originalAction.result_name };
    valid = true;
  }

  return {
    valid,
    amounts,
    resultName,
  };
};

export const stateToNode = (
  settings: NodeEditorSettings,
  state: AirtimeRouterFormState,
): RenderNode => {
  let uuid = createUUID();
  const originalAction = getOriginalAction(settings);
  if (originalAction) {
    uuid = originalAction.uuid;
  }

  const amounts = {};
  state.amounts.forEach((entry: AirtimeTransferEntry) => {
    if (entry.value.amount.trim().length > 0) {
      (amounts as any)[entry.value.code] = Number(entry.value.amount);
    }
  });

  const newAction: TransferAirtime = {
    uuid,
    type: Types.transfer_airtime,
    amounts,
    result_name: state.resultName.value,
  };

  return createWebhookBasedNode(newAction, settings.originalNode, true);
};

export const getOriginalAction = (
  settings: NodeEditorSettings,
): TransferAirtime => {
  const action =
    settings.originalAction ||
    (settings.originalNode.node.actions.length > 0 &&
      settings.originalNode.node.actions[0]);

  if (action.type === Types.transfer_airtime) {
    return action as TransferAirtime;
  }
};
