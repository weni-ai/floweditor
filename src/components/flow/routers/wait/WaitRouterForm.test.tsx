import { RouterFormProps } from 'components/flow/props';
import { Types } from 'config/interfaces';
import { HintTypes, WaitTypes } from 'flowTypes';
import { createMatchRouter, createWaitRouter } from 'testUtils/assetCreators';
import WaitRouterForm from 'components/flow/routers/wait/WaitRouterForm';
import { getTypeConfig } from 'config';
import { render, fireEvent, mock, getCallParams } from 'test/utils';
import * as React from 'react';
import * as utils from 'utils';
import { RenderNode } from 'store/flowContext';
import { getSmartOrSwitchRouter } from 'components/flow/routers/helpers';
import { MEDIA_OPERAND } from 'components/nodeeditor/constants';

mock(utils, 'createUUID', utils.seededUUIDs());

const getRouterFormProps = (
  type: Types,
  originalNode: RenderNode,
): RouterFormProps => {
  return {
    nodeSettings: {
      originalNode,
    },
    typeConfig: getTypeConfig(type),
    assetStore: null,
    updateRouter: vi.fn(),
    onTypeChange: vi.fn(),
    onClose: vi.fn(),
    issues: [],
    helpArticles: {},
  };
};

const colorsWait = getRouterFormProps(
  Types.wait_for_response,
  createMatchRouter(['red', 'green', 'blue']),
);

const audioWait = getRouterFormProps(
  Types.wait_for_audio,
  createWaitRouter(HintTypes.audio, 'Previous Name'),
);

describe(WaitRouterForm.name, () => {
  it('should render a normal wait', () => {
    const { baseElement } = render(<WaitRouterForm {...colorsWait} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('shoud save a normal wait', () => {
    const { getByText } = render(<WaitRouterForm {...colorsWait} />);
    fireEvent.click(getByText('Confirm'));
    expect(colorsWait.updateRouter).toMatchSnapshot();
  });

  it('should render wait for audio with previous name', () => {
    const { baseElement, getByDisplayValue } = render(
      <WaitRouterForm {...audioWait} />,
    );
    getByDisplayValue('Previous Name');
    expect(baseElement).toMatchSnapshot();
  });

  it('should use the right operand for media types', () => {
    const { getByText } = render(<WaitRouterForm {...audioWait} />);
    fireEvent.click(getByText('Confirm'));

    // we should have an operand that looks at attachments
    const [newNode]: [RenderNode] = getCallParams(audioWait.updateRouter);
    const router = getSmartOrSwitchRouter(newNode.node);
    expect(router.operand).toEqual(MEDIA_OPERAND);

    // check the snapshot for the rest of our call
    expect(audioWait.updateRouter).toMatchSnapshot();
  });
});
