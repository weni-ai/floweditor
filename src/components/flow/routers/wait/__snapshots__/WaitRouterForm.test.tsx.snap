// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaitRouterForm shoud save a normal wait 1`] = `
Array [
  Object {
    "inboundConnections": Object {},
    "node": Object {
      "actions": Array [],
      "exits": Array [
        Object {
          "uuid": "6c22884b-0e35-4206-982e-18320691eda9",
        },
      ],
      "router": Object {
        "cases": Array [],
        "categories": Array [
          Object {
            "exit_uuid": "6c22884b-0e35-4206-982e-18320691eda9",
            "name": "All Responses",
            "uuid": "a62e8284-49f8-4292-adbc-d5f6607f85c5",
          },
        ],
        "default_category_uuid": "a62e8284-49f8-4292-adbc-d5f6607f85c5",
        "operand": "@input.text",
        "type": "switch",
        "wait": Object {
          "type": "msg",
        },
      },
      "uuid": "0c8c9239-9b59-473a-a229-d8a9693be270",
    },
    "ui": Object {
      "config": Object {
        "cases": Object {},
      },
      "position": null,
      "type": "wait_for_response",
    },
  },
]
`;

exports[`WaitRouterForm should render a normal wait 1`] = `
<body>
  <div>
    <div
      class="dialog"
    >
      <div
        class="header"
      >
        <div
          class="title_container"
        >
          <div
            class="title u font secondary title-sm black"
          >
            Wait for response
          </div>
          <div
            class="subtitle"
          />
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class=" type_list"
        >
          <div
            class="color-neutral-cloudy"
          >
            When a contact arrives at this point in your flow...
          </div>
          <div>
            <div
              class="select_wrapper"
              data-testid="temba_select_type"
            >
              <div
                data-use-vue-component-wrap=""
                style="all: unset;"
              >
                <div
                  class="unnnic-form sm bold_placeholder"
                  data-passed-props="[object Object]"
                  data-testid="temba_select_input_type"
                  data-v-48c0c241=""
                >
                  <!---->
                  <div
                    class="unnnic-form-input text-input size--sm has-icon-right"
                    data-v-400546b3=""
                    data-v-48c0c241=""
                    mask=""
                    v-bind="[object Object]"
                  >
                    <input
                      class="input-itself input size-sm normal"
                      data-passed-props="[object Object]"
                      data-testid="temba_select_input_type"
                      data-v-400546b3=""
                      data-v-97b44e8c=""
                      iconright="arrow-button-down-1"
                      placeholder="Wait for the contact to respond"
                      size="0"
                      type="text"
                      v-bind="[object Object]"
                    />
                    <!---->
                    <span
                      class="unnnic-icon unnnic-icon__size--sm unnnic-icon-scheme--neutral-cloudy icon-right"
                      data-v-400546b3=""
                      data-v-59b073b4=""
                    >
                      <svg
                        fill="none"
                        viewBox="0 0 40 40"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          class="primary"
                          d="M5.31471 14.0096L19.0432 29.5967C19.1676 29.7264 19.3169 29.8296 19.4822 29.9001C19.6474 29.9707 19.8253 30.007 20.005 30.007C20.1847 30.007 20.3625 29.9707 20.5278 29.9001C20.6931 29.8296 20.8424 29.7264 20.9668 29.5967L34.689 14.0096C34.7987 13.886 34.8825 13.7415 34.9353 13.5849C34.9882 13.4282 35.009 13.2625 34.9965 13.0977C34.984 12.9328 34.9385 12.7721 34.8627 12.6252C34.7869 12.4783 34.6824 12.3481 34.5553 12.2423L32.243 10.2963C31.9852 10.0789 31.6522 9.97163 31.316 9.99759C30.9797 10.0235 30.6671 10.1807 30.4457 10.435L20.2406 22.25C20.2112 22.2841 20.1749 22.3114 20.134 22.3302C20.0931 22.3489 20.0487 22.3587 20.0037 22.3587C19.9588 22.3587 19.9143 22.3489 19.8734 22.3302C19.8326 22.3114 19.7962 22.2841 19.7669 22.25L9.558 10.4338C9.33639 10.1806 9.02374 10.0252 8.68809 10.0013C8.52167 9.9865 8.35397 10.0052 8.19489 10.0563C8.03582 10.1074 7.8886 10.1898 7.76195 10.2988L5.44845 12.2423C5.32061 12.3479 5.21534 12.4781 5.13889 12.6252C5.06245 12.7723 5.0164 12.9332 5.0035 13.0985C4.99099 13.2635 5.01212 13.4294 5.06562 13.586C5.11911 13.7426 5.20386 13.8868 5.31471 14.0096Z"
                          fill="#3B414D"
                        />
                      </svg>
                    </span>
                  </div>
                  <!---->
                </div>
              </div>
              <div
                class="options_wrapper"
                data-testid="temba_select_options_type"
                style="margin-top: 0px; width: 0px; display: none;"
              >
                <div
                  class="option active "
                  data-option-index="0"
                >
                  <span
                    class="name"
                  >
                    Send the contact a message
                  </span>
                </div>
                <div
                  class="option  selected"
                  data-option-index="1"
                >
                  <span
                    class="name"
                  >
                    Wait for the contact to respond
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="2"
                >
                  <span
                    class="name"
                  >
                    Smartly wait for response (New)
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="3"
                >
                  <span
                    class="name"
                  >
                    Send somebody else a message
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="4"
                >
                  <span
                    class="name"
                  >
                    Label the incoming message
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="5"
                >
                  <span
                    class="name"
                  >
                    Add a URN for the contact
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="6"
                >
                  <span
                    class="name"
                  >
                    Add the contact to a group
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="7"
                >
                  <span
                    class="name"
                  >
                    Remove the contact from a group
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="8"
                >
                  <span
                    class="name"
                  >
                    Update the contact
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="9"
                >
                  <span
                    class="name"
                  >
                    Send email
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="10"
                >
                  <span
                    class="name"
                  >
                    Save a result for this flow
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="11"
                >
                  <span
                    class="name"
                  >
                    Call a webhook
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="12"
                >
                  <span
                    class="name"
                  >
                    Call Zapier
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="13"
                >
                  <span
                    class="name"
                  >
                    Enter another flow
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="14"
                >
                  <span
                    class="name"
                  >
                    Start somebody else in a flow
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="15"
                >
                  <span
                    class="name"
                  >
                    Open a ticket with a human agent
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="16"
                >
                  <span
                    class="name"
                  >
                    Send the contact airtime
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="17"
                >
                  <span
                    class="name"
                  >
                    Split by intent
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="18"
                >
                  <span
                    class="name"
                  >
                    Automatic Classify (New)
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="19"
                >
                  <span
                    class="name"
                  >
                    Split by a custom expression
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="20"
                >
                  <span
                    class="name"
                  >
                    Split by contact field
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="21"
                >
                  <span
                    class="name"
                  >
                    Split by a result in the flow
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="22"
                >
                  <span
                    class="name"
                  >
                    Split randomly
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="23"
                >
                  <span
                    class="name"
                  >
                    Split by group membership
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="24"
                >
                  <span
                    class="name"
                  >
                    Split by URN type
                  </span>
                </div>
              </div>
              <div
                class="options_wrapper"
                data-testid="temba_select_expressions_type"
                style="margin-top: 0px; width: 0px; display: none;"
              />
            </div>
          </div>
        </div>
        <div
          class="result_name"
        >
          <div
            class="optional_text_input toggle"
          >
            <span
              class="toggle_link "
              data-spec="toggle-link"
              data-testid="toggle-link"
            >
              Save as
              <div
                data-use-vue-component-wrap=""
                style="all: unset;"
              >
                <span
                  class="unnnic-icon unnnic-icon__size--xs unnnic-icon-scheme--neutral-cleanest"
                  data-passed-props="[object Object]"
                  data-v-59b073b4=""
                >
                  <svg
                    fill="none"
                    viewBox="0 0 40 40"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      class="primary"
                      d="M14.0093 34.6856L29.5962 20.9572C29.726 20.8328 29.8292 20.6835 29.8997 20.5182C29.9702 20.3529 30.0066 20.1751 30.0066 19.9954C30.0066 19.8157 29.9702 19.6379 29.8997 19.4726C29.8292 19.3073 29.726 19.158 29.5962 19.0336L14.0093 5.31149C13.8857 5.20174 13.7412 5.11796 13.5845 5.06514C13.4279 5.01232 13.2622 4.99153 13.0973 5.00401C12.9325 5.01648 12.7718 5.06197 12.6249 5.13776C12.4779 5.21356 12.3477 5.31812 12.242 5.44522L10.296 7.75745C10.0786 8.0153 9.97133 8.34827 9.99728 8.68451C10.0232 9.02076 10.1804 9.33332 10.4347 9.55474L22.2496 19.7598C22.2837 19.7891 22.311 19.8255 22.3298 19.8664C22.3486 19.9072 22.3583 19.9517 22.3583 19.9967C22.3583 20.0416 22.3486 20.0861 22.3298 20.1269C22.311 20.1678 22.2837 20.2042 22.2496 20.2335L10.4385 30.4423C10.3118 30.5523 10.2081 30.6862 10.1333 30.8363C10.0584 30.9865 10.0139 31.1499 10.0023 31.3172C9.98744 31.4836 10.0061 31.6513 10.0572 31.8104C10.1083 31.9695 10.1908 32.1167 10.2997 32.2434L12.247 34.5568C12.3529 34.6836 12.4832 34.7879 12.6301 34.8635C12.7771 34.9391 12.9377 34.9845 13.1024 34.997C13.2672 35.0094 13.4328 34.9887 13.5894 34.9361C13.746 34.8835 13.8905 34.8 14.0143 34.6906L14.0093 34.6856Z"
                      fill="#3B414D"
                    />
                  </svg>
                </span>
              </div>
            </span>
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="buttons"
        >
          <div
            class="buttons_wrapper"
          >
            <div
              class="right_buttons"
            >
              <div
                data-use-vue-component-wrap=""
              >
                <button
                  class="unnnic-button-next unnnic-button-next--size-large unnnic-button-next--ghost unnnic-button-next-scheme--"
                  data-passed-props="[object Object]"
                  data-v-1eef067c=""
                  style="margin-left: 0px; margin-top: 0px;"
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button-next__label"
                    data-v-1eef067c=""
                  >
                     Cancel 
                  </span>
                  <!---->
                </button>
              </div>
              <div
                data-use-vue-component-wrap=""
              >
                <button
                  class="unnnic-button-next unnnic-button-next--size-large unnnic-button-next--primary unnnic-button-next-scheme--undefined"
                  data-passed-props="[object Object]"
                  data-v-1eef067c=""
                  style="margin-top: 0px;"
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button-next__label"
                    data-v-1eef067c=""
                  >
                     Confirm 
                  </span>
                  <!---->
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`WaitRouterForm should render wait for audio with previous name 1`] = `
<body>
  <div>
    <div
      class="dialog"
    >
      <div
        class="header"
      >
        <div
          class="title_container"
        >
          <div
            class="title u font secondary title-sm black"
          >
            Wait for Audio
          </div>
          <div
            class="subtitle"
          />
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class=" type_list"
        >
          <div
            class="color-neutral-cloudy"
          >
            When a contact arrives at this point in your flow...
          </div>
          <div>
            <div
              class="select_wrapper"
              data-testid="temba_select_type"
            >
              <div
                data-use-vue-component-wrap=""
                style="all: unset;"
              >
                <div
                  class="unnnic-form sm bold_placeholder"
                  data-passed-props="[object Object]"
                  data-testid="temba_select_input_type"
                  data-v-48c0c241=""
                >
                  <!---->
                  <div
                    class="unnnic-form-input text-input size--sm has-icon-right"
                    data-v-400546b3=""
                    data-v-48c0c241=""
                    mask=""
                    v-bind="[object Object]"
                  >
                    <input
                      class="input-itself input size-sm normal"
                      data-passed-props="[object Object]"
                      data-testid="temba_select_input_type"
                      data-v-400546b3=""
                      data-v-97b44e8c=""
                      iconright="arrow-button-down-1"
                      placeholder="Wait for an audio recording"
                      size="0"
                      type="text"
                      v-bind="[object Object]"
                    />
                    <!---->
                    <span
                      class="unnnic-icon unnnic-icon__size--sm unnnic-icon-scheme--neutral-cloudy icon-right"
                      data-v-400546b3=""
                      data-v-59b073b4=""
                    >
                      <svg
                        fill="none"
                        viewBox="0 0 40 40"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          class="primary"
                          d="M5.31471 14.0096L19.0432 29.5967C19.1676 29.7264 19.3169 29.8296 19.4822 29.9001C19.6474 29.9707 19.8253 30.007 20.005 30.007C20.1847 30.007 20.3625 29.9707 20.5278 29.9001C20.6931 29.8296 20.8424 29.7264 20.9668 29.5967L34.689 14.0096C34.7987 13.886 34.8825 13.7415 34.9353 13.5849C34.9882 13.4282 35.009 13.2625 34.9965 13.0977C34.984 12.9328 34.9385 12.7721 34.8627 12.6252C34.7869 12.4783 34.6824 12.3481 34.5553 12.2423L32.243 10.2963C31.9852 10.0789 31.6522 9.97163 31.316 9.99759C30.9797 10.0235 30.6671 10.1807 30.4457 10.435L20.2406 22.25C20.2112 22.2841 20.1749 22.3114 20.134 22.3302C20.0931 22.3489 20.0487 22.3587 20.0037 22.3587C19.9588 22.3587 19.9143 22.3489 19.8734 22.3302C19.8326 22.3114 19.7962 22.2841 19.7669 22.25L9.558 10.4338C9.33639 10.1806 9.02374 10.0252 8.68809 10.0013C8.52167 9.9865 8.35397 10.0052 8.19489 10.0563C8.03582 10.1074 7.8886 10.1898 7.76195 10.2988L5.44845 12.2423C5.32061 12.3479 5.21534 12.4781 5.13889 12.6252C5.06245 12.7723 5.0164 12.9332 5.0035 13.0985C4.99099 13.2635 5.01212 13.4294 5.06562 13.586C5.11911 13.7426 5.20386 13.8868 5.31471 14.0096Z"
                          fill="#3B414D"
                        />
                      </svg>
                    </span>
                  </div>
                  <!---->
                </div>
              </div>
              <div
                class="options_wrapper"
                data-testid="temba_select_options_type"
                style="margin-top: 0px; width: 0px; display: none;"
              >
                <div
                  class="option active "
                  data-option-index="0"
                >
                  <span
                    class="name"
                  >
                    Send the contact a message
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="1"
                >
                  <span
                    class="name"
                  >
                    Wait for the contact to respond
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="2"
                >
                  <span
                    class="name"
                  >
                    Smartly wait for response (New)
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="3"
                >
                  <span
                    class="name"
                  >
                    Send somebody else a message
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="4"
                >
                  <span
                    class="name"
                  >
                    Label the incoming message
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="5"
                >
                  <span
                    class="name"
                  >
                    Add a URN for the contact
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="6"
                >
                  <span
                    class="name"
                  >
                    Add the contact to a group
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="7"
                >
                  <span
                    class="name"
                  >
                    Remove the contact from a group
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="8"
                >
                  <span
                    class="name"
                  >
                    Update the contact
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="9"
                >
                  <span
                    class="name"
                  >
                    Send email
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="10"
                >
                  <span
                    class="name"
                  >
                    Save a result for this flow
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="11"
                >
                  <span
                    class="name"
                  >
                    Call a webhook
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="12"
                >
                  <span
                    class="name"
                  >
                    Call Zapier
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="13"
                >
                  <span
                    class="name"
                  >
                    Enter another flow
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="14"
                >
                  <span
                    class="name"
                  >
                    Start somebody else in a flow
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="15"
                >
                  <span
                    class="name"
                  >
                    Open a ticket with a human agent
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="16"
                >
                  <span
                    class="name"
                  >
                    Send the contact airtime
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="17"
                >
                  <span
                    class="name"
                  >
                    Split by intent
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="18"
                >
                  <span
                    class="name"
                  >
                    Automatic Classify (New)
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="19"
                >
                  <span
                    class="name"
                  >
                    Split by a custom expression
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="20"
                >
                  <span
                    class="name"
                  >
                    Split by contact field
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="21"
                >
                  <span
                    class="name"
                  >
                    Split by a result in the flow
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="22"
                >
                  <span
                    class="name"
                  >
                    Split randomly
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="23"
                >
                  <span
                    class="name"
                  >
                    Split by group membership
                  </span>
                </div>
                <div
                  class="option  "
                  data-option-index="24"
                >
                  <span
                    class="name"
                  >
                    Split by URN type
                  </span>
                </div>
              </div>
              <div
                class="options_wrapper"
                data-testid="temba_select_expressions_type"
                style="margin-top: 0px; width: 0px; display: none;"
              />
            </div>
          </div>
        </div>
        <div
          class="result_name"
        >
          <div
            class="optional_text_input "
          >
            <div
              data-testid="Save as result"
            >
              <div
                data-use-vue-component-wrap=""
                style="all: unset;"
                unnnic-input="true"
              >
                <div
                  class="unnnic-form sm"
                  data-passed-props="[object Object]"
                  data-v-48c0c241=""
                  maxlength="64"
                >
                  <p
                    class="unnnic-form__label"
                    data-v-48c0c241=""
                  >
                    Save as result
                  </p>
                  <div
                    class="unnnic-form-input text-input size--sm"
                    data-v-400546b3=""
                    data-v-48c0c241=""
                    mask=""
                    v-bind="[object Object]"
                  >
                    <input
                      class="input-itself input size-sm normal"
                      data-passed-props="[object Object]"
                      data-v-400546b3=""
                      data-v-97b44e8c=""
                      maxlength="64"
                      placeholder="Result"
                      size="13"
                      type="text"
                      v-bind="[object Object]"
                    />
                    <!---->
                    <!---->
                  </div>
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="buttons"
        >
          <div
            class="buttons_wrapper"
          >
            <div
              class="right_buttons"
            >
              <div
                data-use-vue-component-wrap=""
              >
                <button
                  class="unnnic-button-next unnnic-button-next--size-large unnnic-button-next--ghost unnnic-button-next-scheme--"
                  data-passed-props="[object Object]"
                  data-v-1eef067c=""
                  style="margin-left: 0px; margin-top: 0px;"
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button-next__label"
                    data-v-1eef067c=""
                  >
                     Cancel 
                  </span>
                  <!---->
                </button>
              </div>
              <div
                data-use-vue-component-wrap=""
              >
                <button
                  class="unnnic-button-next unnnic-button-next--size-large unnnic-button-next--primary unnnic-button-next-scheme--undefined"
                  data-passed-props="[object Object]"
                  data-v-1eef067c=""
                  style="margin-top: 0px;"
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button-next__label"
                    data-v-1eef067c=""
                  >
                     Confirm 
                  </span>
                  <!---->
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`WaitRouterForm should use the right operand for media types 1`] = `
Array [
  Object {
    "inboundConnections": Object {},
    "node": Object {
      "actions": Array [],
      "exits": Array [
        Object {
          "uuid": "9adce29f-da95-4182-9cb3-0f1e3bf714e5",
        },
      ],
      "router": Object {
        "cases": Array [],
        "categories": Array [
          Object {
            "exit_uuid": "9adce29f-da95-4182-9cb3-0f1e3bf714e5",
            "name": "All Responses",
            "uuid": "6b59e863-2e14-442b-9987-aaa284871b8e",
          },
        ],
        "default_category_uuid": "6b59e863-2e14-442b-9987-aaa284871b8e",
        "operand": "@input",
        "result_name": "Previous Name",
        "type": "switch",
        "wait": Object {
          "hint": Object {
            "type": "audio",
          },
          "type": "msg",
        },
      },
      "uuid": "00dd2423-0a34-4ee2-b97d-839fb2d593ef",
    },
    "ui": Object {
      "config": Object {
        "cases": Object {},
      },
      "position": null,
      "type": "wait_for_response",
    },
  },
]
`;
